galley/pkg/config/processing/snapshotter/snapshot.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshotter
+
+import (
+	"strconv"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/resource"
+	sn "istio.io/istio/pkg/mcp/snapshot"
+)
+
+type snapshot struct {
```

Comment:
maybe call this `snapshotImpl` or `snapshotInstance` to avoid the collision with the `snapshot` package? It's a bit of an annoyance to have the alias everywhere.